lib/src/client/dio_interceptor.dart

```diff
@@ -0,0 +1,226 @@
+import 'dart:async';
+import 'dart:developer';
+import 'dart:io';
+
+import 'package:dio/dio.dart';
+import 'package:intl/intl.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+import 'dart:convert';
+import 'package:requests_signature_dart/src/client/requests_signature_options.dart';
+
+import 'dart:typed_data';
+
+import 'package:uuid/uuid.dart';
+
+/// Interceptor for signing outgoing requests with request signature.
+///
+/// This interceptor signs the outgoing requests with a request signature
+/// before forwarding them to the inner Dio client for processing.
+class RequestsSignatureInterceptor extends Interceptor {
+  Uuid _uuid = const Uuid();
+  final RequestsSignatureOptions _options;
+  final Dio _dioInstance;
+  final ISignatureBodySourceBuilder _signatureBodySourceBuilder;
+  final ISignatureBodySigner _signatureBodySigner;
+  int _clockSkew = 0;
+
+  /// Constructs a new [RequestsSignatureInterceptor].
+  ///
+  /// The [options] parameter specifies the signature options.
+  ///
+  /// Optionally, you can provide custom implementations for
+  /// [signatureBodySourceBuilder] and [signatureBodySigner].
+  RequestsSignatureInterceptor(
+    this._options,
+    this._dioInstance, {
+    ISignatureBodySourceBuilder? signatureBodySourceBuilder,
+    ISignatureBodySigner? signatureBodySigner,
+  })  : _signatureBodySourceBuilder =
+            signatureBodySourceBuilder ?? SignatureBodySourceBuilder(),
+        _signatureBodySigner =
+            signatureBodySigner ?? HashAlgorithmSignatureBodySigner();
+
+  @override
+  Future onRequest(
+      RequestOptions options, RequestInterceptorHandler handler) async {
+    try {
+      print('[✓] onRequest triggered');
+      await _signRequest(options);
+      return handler.next(options);
+    } catch (err, stackTrace) {
+      log('[x] onrequest error');
+      throw RequestsSignatureException('$err\n$stackTrace');
+    }
+  }
+
+  @override
+  void onResponse(Response response, ResponseInterceptorHandler handler) {
+    _validateOptions();
+
+    try {
+      print('[✓] Initial response status code : ${response.statusCode} OK');
+      print('[!] ClockSkew ${_options.clockSkew.inSeconds}');
+
+      if (!_options.disableAutoRetryOnClockSkew &&
+          ((response.statusCode == HttpStatus.unauthorized) ||
+              (response.statusCode == HttpStatus.forbidden)) &&
+          response.headers.value(HttpHeaders.dateHeader) != null) {
+        final rawHeaderDate = response.headers.value(HttpHeaders.dateHeader)!;
+        final serverDate =
+            parseDateString(rawHeaderDate).millisecondsSinceEpoch ~/ 1000;
+        final now = getTime(parseDateString(rawHeaderDate));
```

Comment:
This is functionally different than the .NET implementation. The `getTime` method should not use the `serverDate` as an input because that would affect how the clock skew logic works.